lib/code_corps/messages/conversation_query.ex

```diff
@@ -53,5 +53,16 @@ defmodule CodeCorps.Messages.ConversationQuery do
     |> having([_c, m, _cp], m.initiated_by == "user")
     |> or_having([c, m, cp], m.initiated_by == "admin" and count(cp.id) > 0)
```

Comment:
In response to your question above, this is the query that does it (specifically the `having` and `or_having`), and the documentation above expands on that.